src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -2729,18 +2983,116 @@ object Desugar {
             case w: in.MagicWand => in.ApplyWand(w)(src)
             case e => Violation.violation(s"Expected a magic wand, but got $e")
           }
-        case PExplicitGhostStatement(actual) => stmtD(ctx)(actual)
+        case p: PClosureImplProof => closureImplProofD(ctx)(p)
+        case PExplicitGhostStatement(actual) => stmtD(ctx, info)(actual)
         case _ => ???
       }
     }
 
+    private def closureImplProofD(ctx: FunctionContext)(proof: PClosureImplProof): Writer[in.Stmt] = {
```

Comment:
Please, add more documentation of what is happening inside this method. On first glance, the body seems fine to me, but we need proper documentation so that it is maintainable for us.

Reply:
Done